lib/compiler-llvm/src/abi/aarch64_systemv.rs

```diff
@@ -83,34 +83,51 @@ impl Abi for Aarch64SystemV {
 
         Ok(match sig.results() {
             [] => (
-                intrinsics
-                    .void_ty
-                    .fn_type(&param_types.collect::<Result<Vec<_>, _>>()?, false),
+                intrinsics.void_ty.fn_type(
+                    param_types
+                        .map(|v| v.unwrap().into())
```

Comment:
The semantics here have changed from returning an error value if any of these things fail to just panicking. You can probably preserve the collect to `Result<Vec<_>, _>` behavior if you want with something like

```param_types.map(|v| v.and_then(|v| Ok(v.into()))).collect::<Result<Vec<BasicMetadataTypeEnum>, _>()?```